payara-common/src/main/java/fish/payara/arquillian/container/payara/CommonPayaraManager.java

```diff
@@ -116,6 +131,10 @@ public ProtocolMetaData deploy(Archive<?> archive) throws DeploymentException {
         }
 
         final String archiveName = archive.getName();
+        final String deploymentName = createDeploymentName(prependRandomToDeploymentName ?
+                String.format("r%d-%s", random.nextInt(1000), archiveName) : archiveName);
+
+        log.log(Level.INFO, "Deploying {0}", new Object[] { deploymentName });

```

Comment:
With this solution there's still 0.1% chance of name clash. If this is solely for situation of running within single JVM process a static AtomicInteger would serve the usecase better.

Reply:
Or use UUID.randomUUID() :-)

Reply:
yes, that's a better solution :)